src/fits.cc

```diff
@@ -87,6 +87,10 @@ struct FitsType<char> {
     static int const CONSTANT = TSTRING;
 };
 template <>
+struct FitsType<signed char> {
+    static int const CONSTANT = TSBYTE;
+};
+template <>
 struct FitsType<unsigned char> {
     static int const CONSTANT = TBYTE;
 };
```

Comment:
Any reason we don't specifically also specialize for `std::uint8_t`? I know it maps on most platforms, but it doesn't have to.

Reply:
I'm not sure. I am copying existing code which may be based on this entry in the CFITSIO documentation: https://heasarc.gsfc.nasa.gov/docs/software/fitsio/quick/node9.html. It is true that the entry refers to images (it was the only such mention I found).

(begin quote)
The datatype parameter specifies the datatype of the C array in the program, which need not be the same as the datatype of the FITS image itself. If the datatypes differ then CFITSIO will convert the data as it is read or written. The following symbolic constants are allowed for the value of datatype:
  TBYTE     unsigned char
  TSBYTE    signed char
  TSHORT    signed short
  TUSHORT   unsigned short
  TINT      signed int
  TUINT     unsigned int
  TLONG     signed long
  TLONGLONG signed 8-byte integer
  TULONG    unsigned long
  TFLOAT    float
  TDOUBLE   double
(end quote)

Reply:
Yes, as @r-owen said, this is limited by what cfitsio supports.